optimizely/optimizely_config.py

```diff
@@ -0,0 +1,214 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+
+
+class OptimizelyConfig(object):
+    def __init__(self, revision, experiments_map, features_map):
+        self.revision = revision
+        self.experiments_map = experiments_map
+        self.features_map = features_map
+
+
+class OptimizelyExperiment(object):
+    def __init__(self, id, key, variations_map):
+        self.id = id
+        self.key = key
+        self.variations_map = variations_map
+
+
+class OptimizelyFeature(object):
+    def __init__(self, id, key, experiments_map, variables_map):
+        self.id = id
+        self.key = key
+        self.experiments_map = experiments_map
+        self.variables_map = variables_map
+
+
+class OptimizelyVariation(object):
+    def __init__(self, id, key, feature_enabled, variables_map):
+        self.id = id
+        self.key = key
+        self.feature_enabled = feature_enabled
+        self.variables_map = variables_map
+
+
+class OptimizelyVariable(object):
+    def __init__(self, id, key, variable_type, value):
+        self.id = id
+        self.key = key
+        self.type = variable_type
+        self.value = value
+
+
+class OptimizelyConfigService(object):
+    """ Class encapsulating methods to be used in creating instance of OptimizelyConfig. """
+
+    def __init__(self, project_config):
+        """
+        Args:
+            project_config ProjectConfig
+        """
+        self.experiments = project_config.experiments
+        self.feature_flags = project_config.feature_flags
+        self.groups = project_config.groups
+        self.revision = project_config.revision
+
+        self._create_lookup_maps()
+
+    def get_config(self):
+        """ Returns instance of OptimizelyConfig
+
+        Returns:
+            Optimizely Config instance.
+        """
+
+        experiments_key_map, experiments_id_map = self._get_experiments_maps()
+        features_map = self._get_features_map(experiments_id_map)
+
+        return OptimizelyConfig(self.revision, experiments_key_map, features_map)
+
+    def _create_lookup_maps(self):
+        """ Creates lookup maps to avoid redundant iteration of config objects.  """
+
+        self.exp_id_to_feature_map = {}
+        for feature in self.feature_flags:
+            for id in feature['experimentIds']:
+                self.exp_id_to_feature_map[id] = feature
+
+        self.feature_key_variable_key_to_variable_map = {}
+        self.feature_key_variable_id_to_variable_map = {}
+        for feature in self.feature_flags:
```

Comment:
am unable to find the difference between this `feature_flags` loop and at line 86. Can't we use one loop only?